external_codes/github_actions/auto-rebase.sh

```diff
@@ -127,4 +127,20 @@ git checkout -b fork/$HEAD_BRANCH fork/$HEAD_BRANCH
 git rebase origin/$BASE_BRANCH
 
 # push back
-git push --force-with-lease fork fork/$HEAD_BRANCH:$HEAD_BRANCH
\ No newline at end of file
+git push --force-with-lease fork fork/$HEAD_BRANCH:$HEAD_BRANCH
```

Comment:
what happens if this push fails.  Say an attacker knows the what the hash of the rebase will be and already pushed it.  Will this action still continue and use the COMMIT_SHA?

Reply:
I can look into creating a bot account and signing commits as outlined in this article:
https://httgp.com/signing-commits-in-github-actions/

This would essentially be adding an additional unknown salt to the commit hash.

But also according to this article git is aware of and actively monitor commits for collisions:
https://github.blog/2017-03-20-sha-1-collision-detection-on-github-com/

Reply:
The script has "set -e" specified at the top, so the script will exit if this command exits with a non-zero error code.  And not submit the following approval code. Under what conditions would this push fail and what state would the PR be left in?